papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,564 @@
+
+**Dynamic Workspaces:**
+The **signac-flow** package can now handle workspaces where jobs are created as the result of operations on other jobs.
+This is crucial for optimization workflows and iteratively sampling parameter spaces, and allows projects to become more automated with some data points only run if a prior condition on another data point is reached.
+
+Executing complex workflows via groups and aggregation
+------------------------------------------------------
+
+.. figure:: signac-flow_aggregation-groups-bundling.pdf
+    :align: center
+    :scale: 90 %
+    :figclass: w
+
+    Aggregation, groups, and bundling allow users to build complex workflows.
+    The features are orthogonal, and can be used in any combination.
+    Aggregation enables one operation or group to act on multiple jobs.
+    Groups allow users to combine multiple operations into one, with dependencies among operations resolved at run time.
+    Bundling helps users efficiently leverage HPC schedulers by submitting multiple commands in the same script, to be executed in serial or parallel. :label:`workflow`
+
+Two new concepts in **signac-flow** provide users with significantly more power to implement complex workflows: groups and aggregation.
+A related third concept – bundling – which is not new, also provides flexibility to users in their workflows, but exclusively affects scheduler submission, not workflow definition.
+Figure :ref:`workflow` show a graphical illustration of the three concepts.
+
+As the names of both groups and aggregation imply, the features enable the "grouping" or "aggregating" of existing concepts: operations in the case of groups and jobs in the case of aggregates.
+The conceptual model of **signac-flow** builds on **signac**'s notions of the ``Project`` and ``Job`` (the unit of the data space) through a ``FlowProject`` class that adds the ability to define and execute operations (the unit of a workflow) that act on jobs.
+Operations are Python functions or shell commands that act on a job within the data space, and are defined using Python decorator syntax.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject
+
+    @FlowProject.operation
+    @Flowproject.post.true("initialized")
+    def initialize(job):
+        # perform necessary initialize steps
+        # for simulation
+        job.doc.initialized == True
+
+    if __name__ == "__main__":
+        FlowProject().main()
+
+When this project is run using **signac-flow**'s command line API (``python project.py run``), the current state point is prepared for simulation.
+Operations can have preconditions and postconditions that define their eligibility.
+All preconditions must be met in order for a operation to be eligible for a given job.
+If any postcondition is met, that indicates an operation is complete (and thus ineligible).
```

Comment:
```suggestion
If all postconditions are met, that indicates an operation is complete (and thus ineligible).
```

I thought all postconditions must be true for an operation to be complete. In the case I tried with multiple postconditions, if one postcondition is not met and one postcondition is met, then the operation still runs. Did I mistunderstand?

Reply:
this is my understanding as well.